libs/designsystem/header-actions/src/header-actions.component.ts

```diff
@@ -0,0 +1,117 @@
+})
+export class HeaderActionsComponent implements AfterViewInit {
+  /*
+   * When the consumer sets the visibleActions input, we run all our initialization logic.
+   * The guard for this.buttons exist because when the component is instantiated,
+   * the ContentChild query has not yet run, and all the below functions assume that
+   * the buttons array is defined. When undefined, we defer the initialization to ngAfterViewInit().
+   */
+  @Input() set visibleActions(value: number) {
+    this._visibleActions = value;
+
+    if (!this.buttons) return;
+    this.initializeCollapsing();
+  }
+
+  @HostBinding('class')
+  @Input()
+  placement: 'left' | 'right' = 'left';
+
+  @ViewChild(DropdownComponent, { read: ElementRef }) dropdown!: ElementRef;
+  @ContentChildren(ButtonComponent, { read: ElementRef }) buttons!: ElementRef<HTMLButtonElement>[];
+  @ViewChild('hiddenLayer', { read: ElementRef }) hiddenLayer!: ElementRef<HTMLElement>;
+  @ViewChild('visibleLayer', { read: ElementRef }) visibleLayer!: ElementRef<HTMLElement>;
+
+  _collapsedActions: string[] = [];
+  _visibleActions: number;
+
+  private dropdownTextToButtonMap: Map<string, HTMLButtonElement>;
+
+  constructor(private renderer: Renderer2) {}
+
+  ngAfterViewInit(): void {
+    this.initializeCollapsing();
+  }
+
+  initializeCollapsing() {
+    if (this.buttons.length <= 2) return;
+    this.moveButtons();
+    this.populateDropdown();
+    this.toggleDropdown();
+  }
+
+  moveButtons() {
+    this.buttons.forEach((button, index) => {
+      if (index > this._visibleActions - 1) {
+        this.renderer.appendChild(this.hiddenLayer.nativeElement, button.nativeElement);
+      } else {
+        this.renderer.appendChild(this.visibleLayer.nativeElement, button.nativeElement);
+      }
+    });
+  }
+
+  toggleDropdown() {
+    if (this.hiddenLayer.nativeElement.childElementCount === 0) {
+      this.renderer.setStyle(this.dropdown.nativeElement, 'display', 'none');
+    } else {
+      this.renderer.setStyle(this.dropdown.nativeElement, 'display', 'inline-block');
+    }
+  }
+
+  populateDropdown() {
+    /*
+     * This function maps extracts the button text of all hidden buttons
+     * and updates the array used to populate the dropdown with items.
+     * It also maps the extracted text to the actual button element, for use when
+     * firing the matching buttons click-event in onDropdownActionSelect
+     */
+
+    this.dropdownTextToButtonMap = new Map<string, HTMLButtonElement>();
+    this._collapsedActions = [];
+
+    const hiddenButtons = Array.from(
+      this.hiddenLayer.nativeElement.children
+    ) as HTMLButtonElement[];
+
+    hiddenButtons.forEach((button) => {
+      const buttonText = button.textContent.trim();
+      this._collapsedActions.push(buttonText);
+      this.dropdownTextToButtonMap.set(buttonText, button);
+    });
+  }
+
+  onDropdownActionSelect(action) {
+    const selectedAction = this.dropdownTextToButtonMap.get(action);
```

Comment:
Consider if it might be too brittle to rely on the text alone, compared to using let's and index instead? What happens if 2 buttons has the same text but different icons? Maybe not likely...